### DISCUSSION TITLE
Font Scaling on Small Viewport (Blurriness Issue)

### DISCUSSION THREAD

**Discussion by owlsowlsowlsowlsowlsowlsowlsowlsowls** [dialogic-godot/dialogic#2359] [Q&A]
My viewport is 320x180 for a pixel art game.

I've made a custom theme to try to fix the issue.
![image](https://github.com/user-attachments/assets/c1664e9f-e969-4463-b124-256448f66036)
Default settings.
![image](https://github.com/user-attachments/assets/e1dd951e-ebd5-4472-ac64-b0cb85bc30ef)
Changing font size from default (15) to 5.

I have nearest texture filter on as well as nearest decals. More settings   vvv
![image](https://github.com/user-attachments/assets/24365146-c222-40ad-b038-980758a97947)

Is my only choice to scale everything up to match the Dialogic window or is there actually a way to scale down Dialogic?

**Comment by timberdive** [comment]
I have the same issue, waiting eagerly for a solution

**Comment by timberdive** [comment]
Hey so I found a solution for this! In my pixel game I initially used the "scale" setting to zoom in on my pixel game. Now I changed the scale back to one  and in the camera 2d node I set zoom to 6. Hope this fixes the issue for you as well!

**Comment by Jowan-Spooner** [comment]
Hey, I just saw this. Here are some tips for pixel-art games.

The setup that works best for me is this:
Project Settings:
- Display/Window/Size
    - Viewport Width: `320`
    - Viewport Height: `180`
- Display/Window/Stretch
    - Mode: `viewport`
    - Scale Mode: `integer`
- Rendering/Textures
    - CanvasTextures/DefaultTextureFilter: `Nearest`
    
Most importantly **download a pixel-art font** ([kenney has some free ones](https://kenney.nl/assets/kenney-fonts)). In godot double-click the font to open the font importer window. Set `Antialiasing` to `None`.

Create a new dialogic style, change the box size to fit into the game size.
Setup the style to use the font. If you do so on the base layer, the other layers can use this font automatically.
Change the font size to match the actual font size (e.g. 16 pixels for a 16 pixel high font.

This results in the best looking fonts I think, though I have personally never made a big pixel-art project and only played around with it.

**Reply by timberdive** [reply]
I found that setting the stretch mode to canvas item solved the issue for me.
